### ISSUE TITLE
Possibly support Multi-master and/or Mirror mode

### ISSUE THREAD

**Issue by bodgit** [bodgit/puppet-openldap#10]

**Comment by bodgit** [comment]
ITS#7274

**Comment by Yyuzu** [comment]
Hi,

I needed this feature, so I added it quickly.
A patch is attached, it's quite simple and I hope I didn't make any errors in there.

[puppet-openldap-mirrormode.diff.txt](https://github.com/bodgit/puppet-openldap/files/469470/puppet-openldap-mirrormode.diff.txt)

**Comment by bodgit** [comment]
In theory, what you've provided describes the end state correctly, however the documentation is not very clear and I've seen conflicting reports/howtos that state you should first set olcServerID to just a number before setting it to a list of ID + URL pairs. Also, in the ITS ticket I posted as a reminder to myself, you supposedly should set the server ID _after_ the `syncprov` overlay has been enabled on the database so on a fresh install I'd need to defer setting the server ID until the overlay has been enabled which would perhaps need multiple Puppet runs which I'd like to avoid.

Sadly I don't think it's as simple as just adding those extra settings which is why I've not done it yet. I'd love to find a canonical document that explains the correct steps and more importantly _why_ it has to be done that way.

**Comment by Yyuzu** [comment]
You're right, official documentation indeed states it should have a standalone numeric ServerID first :
http://www.openldap.org/doc/admin24/replication.html
I noticed that but since we're still in project mode I took my chances anyways, but I'm no OpenLDAP expert... I don't know why it is required, or if it is at all in fact, but I'd be interested to know too !

Do you think it's possible to achieve such configuration in one puppet run ? Maybe with multiple reload calls and dependencies ?
